src/terminal-parser.cpp

```diff
@@ -61,10 +61,11 @@ namespace librealsense
             data_vec.insert(data_vec.begin(), data.begin(), data.end());
             return data_vec;
         }
-        else
+        else if (std::all_of(response.begin() + 1, response.end(), [](int x) { return x == 0; })) // if the response contains only the opcode, the response is empty
         {
-            return response;
+            return {};
         }
+        return response;
```

Comment:
Isn't this changing the logic? previously the opcode was returned, now only an empty buffer

Reply:
I also would like the opcode to return with some success message maybe?